addons/adyen_platforms/models/adyen_account.py

```diff
@@ -0,0 +1,711 @@
+        adyen_account_id.with_context(update_from_adyen=True).write({
+            'adyen_uuid': response['adyen_uuid'],
+            'proxy_token': response['proxy_token'],
+        })
+
+        # A default payout is created for all adyen accounts
+        adyen_account_id.env['adyen.payout'].with_context(update_from_adyen=True).create({
+            'code': response['adyen_response']['accountCode'],
+            'adyen_account_id': adyen_account_id.id,
+        })
+        return adyen_account_id
+
+    def write(self, vals):
+        res = super(AdyenAccount, self).write(vals)
+        if not self.env.context.get('update_from_adyen'):
+            self._adyen_rpc('update_account_holder', self._format_data())
+        return res
+
+    def unlink(self):
+        for adyen_account_id in self:
+            adyen_account_id._adyen_rpc('close_account_holder', {
+                'accountHolderCode': adyen_account_id.account_holder_code,
+            })
+        return super(AdyenAccount, self).unlink()
+
+    @api.model
+    def action_create_redirect(self):
+        '''
+        Accessing the FormView to create an Adyen account needs to be done through this action.
+        The action will redirect the user to accounts.odoo.com to link an Odoo user_id to the Adyen
+        account. After logging in on odoo.com the user will be redirected to his DB with a token in
+        the URL. This token is then needed to create the Adyen account.
+        '''
+        if self.env.company.adyen_account_id:
+            # An account already exists, show it
+            return {
+                'name': _('Adyen Account'),
+                'view_mode': 'form',
+                'res_model': 'adyen.account',
+                'res_id': self.env.company.adyen_account_id.id,
+                'type': 'ir.actions.act_window',
+            }
+        return_url = url_join(self.env['ir.config_parameter'].sudo().get_param('web.base.url'), 'adyen_platforms/create_account')
+        onboarding_url = self.env['ir.config_parameter'].sudo().get_param('adyen_platforms.onboarding_url')
+        return {
+            'type': 'ir.actions.act_url',
+            'url': url_join(onboarding_url, 'get_creation_token?return_url=%s' % return_url),
+        }
+
+    def action_show_transactions(self):
+        return {
+            'name': _('Transactions'),
+            'view_mode': 'tree,form',
+            'domain': [('adyen_account_id', '=', self.id)],
+            'res_model': 'adyen.transaction',
+            'type': 'ir.actions.act_window',
+            'context': {'group_by': ['adyen_payout_id']}
+        }
+
+    def _upload_photo_id(self, document_type, content, filename):
+        self._adyen_rpc('upload_document', {
+            'documentDetail': {
+                'accountHolderCode': self.account_holder_code,
+                'documentType': document_type,
+                'filename': filename,
+            },
+            'documentContent': content,
```

Comment:
json.dumps does not work on bytes object, need to convert into string
```suggestion
            'documentContent': content.decode(),
```